10539 export.rst doc edit (#10565)
bokeh/bokeh: sphinx/source/docs/user_guide/export.rst

.. _userguide_export:

Exporting plots
===============

Additional dependencies
-----------------------

> You may need additional dependencies to use the |export| functions.

Reviewer: ```suggestion
You may need to install additional dependencies to use the |export| functions.
```

You can install these dependencies in several ways, including the
following:

• Using Conda:

.. code-block:: sh

    conda install selenium geckodriver firefox -c conda-forge

• Using ChromeDriver and the Chromium browser or a suitable alternative.

.. _userguide_export_png:

> PNG generation

Reviewer 2: I think this should be "Exporting PNG images"

.. _userguide_export:

Exporting plots
===============

Additional dependencies
-----------------------

You may need additional dependencies to use the |export| functions.
You can install these dependencies in several ways, including the
following:

> • Using Conda:

Reviewer: They might render in a way, because Sphinx just puts the asci character into the html code. However, this is not a good way to generate lists with Sphinx - you should definitely use asterisks as described in the link to the spinx docs.

> • Using Conda:

Reviewer 2: Right, sorry I was not specific, I meant would they be rendered as actual `<li>` items as an asterisk would be

> • Using Conda:

Author: Okay, noted. Assumed it would be okay to use this character because of how [bullet lists](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#bullet-lists) are described in the reStructuredText Markup Specification.

> • Using Conda:

Reviewer: Using asterisks also ensures a certain level of consistency throughout the various elements of Bokeh's docs. There are often different ways to do one and the same thing with ReST. When in doubt, I would generally follow the recommendations in the Sphinx documentation.

.. _userguide_export_svg:

> SVG generation

Reviewer 2: I think this should be "Exporting SVG images"

--------------

> Bokeh can also replace HTML5 Canvas plot output with an Scalable Vector

Reviewer: ```suggestion
Bokeh can also replace the HTML5 Canvas plot output with a Scalable Vector
```

Graphics (SVG) element that can be edited in image editing programs such
as Adobe Illustrator and/or converted to a PDF.

The SVG output isn't as performant as the default Canvas backend when it comes
to rendering a large number of glyphs or handling lots of user interactions like
panning.

To create an SVG with a transparent background, set the ``Plot.background_fill_color``

> and ``Plot.border_fill_color`` properties to ``None``, same as with PNG output.

Reviewer 2: ```suggestion
and ``Plot.border_fill_color`` properties to ``None``.
```
I'm not sure the callback reference to PNG adds anything.

Limitations
~~~~~~~~~~~

You can't create a single SVG for a layout of plots because each plot will
produce its own distinct SVG element. You can, however, download an SVG plot

> using a SaveTool from the toolbar although an SVG export of a plot with a

Reviewer: very long sentence with lots of modifiers. Maye somthing like: "... using a SaveTool from the toolbar. If you use this option to create an SVG export and your plot contains a toolbar, keep in mind that the exported file will have a blank area ..."

Author: Okay, yeah. This one is a bit heavy-handed. Broke it up and simplified.

> will have a blank area where the toolbar used to be.

Reviewer 2: I think this entire para needs more attention. These two cases don't really have anything to do with one another, but it reads (since the original) like somehow the savetool is a solution to the first issue with layouts. 

Suggest that the last two sentences be combined:

> You can download an SVG plot using a SaveTool from the toolbar (note that the exported file
will have a blank area where the toolbar was).

And I think this should be separated from the first sentence as well, either two very short one-sentence para's or turn in to a bulleted list maybe?

@tcmetzger about this:

> You can't create a single SVG for a layout of plots because each plot will produce its own distinct SVG element.

I wanted to see if you have any suggestion for different wording. This is a limitation of the library, but in this context the "You can't ..." reads to me like "You aren't allowed" TBH I prefer direct statements like "It is not possible to..." in cases where things are.... not possible. What is the compromise update here?

Reviewer 2: Oh I see, that original "Alternatively,..." looks to be copy-pasta from another section. It was definitely badly worded from the start.

Reviewer: Well, "it is not possible" sounds pretty good to me! Or you could turn around the sentence a bit: "Each plot is its own SVG element. Therefore, it is not possible to automatically create a single SVG combining all plots in a gridplot layout."

    # option two
    plot.output_backend = "svg"

> Exporting an SVG image

Reviewer 2: This whole section is very muddled (has been from the start). I think the "SVG Generation" headline should become "Exporting SVG images" and end with `export_pngs`. Then subsection for "limitations" and maybe finally a section about saving directly from browsers (e.g. with the bookmarlket)
